src/sql/azdata.proposed.d.ts

```diff
@@ -1801,5 +1828,66 @@ declare module 'azdata' {
 
 		export interface Dialog extends LoadingComponentBase {
 		}
+
+		/**
+		 * Provides dialog options to customize modal dialog content and layout
+		 */
+		export interface ICustomDialogOptions {
+			/**
+			 * Severity Level to identify icon of modal dialog.
+			 */
+			severity: MessageLevel;
+			/**
+			 * Title of modal dialog header.
+			 */
+			headerTitle: string;
+			/**
+			 * Message text to show on dialog.
+			 */
+			message: string;
+			/**
+			 * (Optional) Detailed message, e.g stack trace of error.
+			 */
+			messageDetails?: string;
+			/**
+			 * (Optional) List of custom actions to include in modal dialog alongwith a 'Cancel' button.
+			 * If custom 'actions' are not provided, 'OK' and 'Cancel' buttons will be shown by default.
+			 */
+			actions?: IDialogAction[];
+			/**
+			 * (Optional) If provided, instruction text is shown in bold below message.
+			 */
+			instructionText?: string;
+			/**
+			 * (Optional) If provided, appends read more link after instruction text.
+			 */
+			readMoreLink?: string;
+		}
+
+		/**
+		 * An action that will be rendered as a button on the dialog.
+		 */
+		export interface IDialogAction {
+			/**
+			 * Identifier of action.
+			 */
+			id: string;
+			/**
+			 * Label of Action button.
+			 */
+			label: string;
+			/**
+			 * Defines if button styling and focus should be based on primary action.
+			 */
+			isPrimary: boolean;
+			/**
+			 * Defines if dialog should be closed on action event.
+			 */
+			closeDialog: boolean;
```

Comment:
what is the use case for this?  I am not sure how this will work with the openDialog API. it is expecting an action id as return result, but if you are not closing the dialog, other actions could be triggered and won't be handled.

Reply:
This is added for Diagnostic RP, when an action to run diagnostics will not close the dialog, but update the diagnostic results in the same dialog.

Reply:
it looks like the flag is not yet used in this PR. I think my previous question was not answered, could you please describe the desired workflow?

Reply:
Will remove it and add later